js/dydx.js

```diff
@@ -0,0 +1,470 @@
+        //         nextFundingRate: '-0.0000519113',
+        //         nextFundingAt: '2022-01-25T14:00:00.000Z',
+        //         minOrderSize: '0.001',
+        //         type: 'PERPETUAL',
+        //         initialMarginFraction: '0.05',
+        //         maintenanceMarginFraction: '0.03',
+        //         volume24H: '865742935.580100',
+        //         trades24H: '142258',
+        //         openInterest: '5290.9970',
+        //         incrementalInitialMarginFraction: '0.01',
+        //         incrementalPositionSize: '1.5',
+        //         maxPositionSize: '170',
+        //         baselinePositionSize: '9',
+        //         assetResolution: '10000000000',
+        //         syntheticAssetId: '0x4254432d3130000000000000000000'
+        //       },
+        //       ...
+        //    }
+        // }
+        //
+        const result = [];
+        const markets = this.safeValue (marketsResponse, 'markets');
+        const keys = Object.keys (markets);
+        const settleId = 'USDC'; // https://docs.dydx.exchange/#margin
+        for (let i = 0; i < keys.length; i++) {
+            const marketKey = keys[i];
+            const market = markets[marketKey];
+            const id = this.safeString (market, 'market', marketKey);
+            const active = this.safeString (market, 'status') === 'ONLINE';
+            const baseId = this.safeString (market, 'baseAsset');
+            const quoteId = this.safeString (market, 'quoteAsset');
+            const base = this.safeCurrencyCode (baseId);
+            const quote = this.safeCurrencyCode (quoteId);
+            const settle = this.safeCurrencyCode (settleId);
+            const symbol = this.symbolDefine (base, quote, settle);
+            const stepSize = this.safeNumber (market, 'stepSize');
+            const tickSize = this.safeNumber (market, 'tickSize');
+            const minOrderSize = this.safeNumber (market, 'minOrderSize');
+            const type = this.safeString (market, 'type');
+            const spot = false;
+            const future = false;
+            const swap = (type === 'PERPETUAL');
+            const option = false;
+            const contract = future || swap;
+            const entry = {
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'settle': settle,
+                'baseId': baseId,
+                'quoteId': quoteId,
+                'settleId': settleId,
+                'type': type,
+                'spot': spot,
+                'margin': false,
+                'future': future,
+                'swap': swap,
+                'option': option,
+                'active': active,
+                'contract': contract,
+                'linear': undefined,
+                'inverse': undefined,
+                // 'maker': maker, // TODO: I suppose this should be set from base, depending on global market/taker values
```

Comment:
That is correct, we don't need them here, if the fees are uniform for all markets.